Autocompile

samuel-massinon opened the pull request:
This was the best way I could find to include the python path in the build, if there's a better way that allows us to not have hard coded string in the file and for it not to force everything to have everything in `~/.pyenv/versions/2.7.9/` please let me know.

Commit:
Made autocompile for matpy

Commit:
Forgot to save my file before pushing

iamed2 commented:
`PYNAME` and the -`D`s should go in `CFLAGS` like the example I sent you. All other values should be separate arguments to `mex`.

iamed2 commented:
Instead of assuming people are using `pyenv`, you should discover the path of people's python installs through `which python`/`where python` and use `readlink`/(this solution: http://stackoverflow.com/a/27407405) to get the path of the actual executable if it's a symlink.

Commit:
Added cflags and put cflags, pyname and pypath together
Can now get the custom python path from the file .matpyrc
Updated the readme

samuel-massinon commented:
@iamed2 The problem I'm getting for your last comment is that matpy can't seem to find pyenv without me telling it explicitly. If I say `which python` in unix I get the right one, but in MATLAB when I enter `system('which python')` I get the wrong one. The Makefile uses `PYPATH` in the `.bash_profile` to point to the right python, but MATLAB has no knowledge of `PYPATH`. I'll keep looking for a solution. Any help would be great too. :)

iamed2 commented on README.md:
```diff
@@ -208,3 +208,7 @@ If your last line after compiling is,
+
+### Pointing to the right python
+
+By default, matpy will build with the systems default python. If you want it to reference a specific python, you'll need to make a file `~/.matpyrc` and set the python you want matlab to use (and only that string), for example I have `/.pyenv/versions/2.7.9/bin/python` as the only line in my `~/.matpyrc` file.
```
This should be the full path (including the preceding tilde)

Commit:
Cleaned up the files
Removed once most once used variables
Made the compiler not assume users are using python 2.7
Compiler will default to system default python if custom python is not
found.
Added to the README python 2.7 is supported and tested for

Commit:
Made the include path check to look only for directories

Commit:
Removed the folder code
